plugins/state_history_plugin/include/eosio/state_history_plugin/session.hpp

```diff
@@ -250,15 +250,24 @@ class session final : public session_base {
                      block_to_send->blocks_result_base.this_block  = {self.current_blocks_request.start_block_num, *this_block_id};
                      if(const std::optional<chain::block_id_type> last_block_id = self.get_block_id(self.next_block_cursor - 1))
                         block_to_send->blocks_result_base.prev_block = {self.next_block_cursor - 1, *last_block_id};
-                     if(chain::signed_block_ptr sbp = get_block(self.next_block_cursor); sbp && self.current_blocks_request.fetch_block)
-                         block_to_send->blocks_result_base.block = fc::raw::pack(*sbp);
+                     if (self.current_blocks_request.fetch_block) {
+                        if (chain::signed_block_ptr sbp = get_block(*this_block_id)) {
+                           block_to_send->blocks_result_base.block = fc::raw::pack(*sbp);
+                        } else {
+                           fc_wlog(logger, "Unable to retrieve block ${bn} : ${id}",
```

Comment:
This treats the blog specially, and I can't really think of a reason to do so. Either all 4 logs should warn in this case, or none should, to maintain consistency across the 4.

But a log message here just seems ripe for a log flood. For example if someone has `block-log-retain-blocks` set low, a trace log from genesis, and requests block+trace from genesis, I don't think we need to print millions of `wlog`s when the missing block _is already conveyed_ in the still usable response (the block's `optional<bytes>` is unset)

In other words the client already has an explicit visible indicator of this error (on any of the 4 logs), so unless we have a good strong reason to add the log I'd rather leave it without a log and avoid bloating this out more.